.github/workflows/publish.yml

```diff
@@ -17,42 +16,55 @@ permissions:
   contents: read
 
 jobs:
-  # calls our general CI workflow (tests, build docs, etc.)
+  # calls our general CI workflows (tests, coverage, profile, etc.)
   tests:
     uses: ./.github/workflows/CI.yml
     # needed for the coverage action
     permissions:
       contents: write
       pull-requests: write
+  # calls our docs workflow (build docs, check broken links, lighthouse)
+  docs:
+    uses: ./.github/workflows/docs.yml
 
   build-package:
     name: "Build & verify PST package"
-    needs: [tests] # require tests to pass before deploy runs
+    # require tests and docs to pass before building the package
+    needs: [tests, docs]
     runs-on: ubuntu-latest
     steps:
       - name: "Checkout repository 🛎"
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # 4.2.2
+        with:
+          persist-credentials: false
 
       - name: "Setup CI environment 🛠"
-        uses: ./.github/actions/set-dev-env
+        uses: pydata/pydata-sphinx-theme/.github/actions/set-dev-env@01731d0cc57768b9eff1c97f38909932ecd7e7d1
```

Comment:
The `set-dev-env` action was documented to not be used for the release process but is used for building the package?

Reply:
Good catch, I planned to remove it altogether and just setup the env from scratch here, and I forgot.